lib/rucio/core/transfer.py

```diff
@@ -400,24 +400,24 @@ def ensure_db_sources(
 
 @transactional_session
 def set_transfers_state(
-        transfers,
-        state: "RequestState",
+        transfers: "Sequence[DirectTransfer]",
+        state: RequestState,
         submitted_at: datetime.datetime,
         external_host: str,
         external_id: str,
         transfertool: str,
         *,
         session: "Session",
-        logger
-):
+        logger: LoggerFunction = logging.log,
+) -> None:
     """
     Update the transfer info of a request.
     :param transfers:  Dictionary containing request transfer info.
     :param session:    Database session to use.
     """
 
     logger(logging.INFO, 'Setting state(%s), transfertool(%s), external_host(%s) and eid(%s) for transfers: %s',
-           state.name, transfertool, external_host, external_id, ', '.join(t.rws.request_id for t in transfers))
+           state.name, transfertool, external_host, external_id, ', '.join(str(t.rws.request_id) for t in transfers))
```

Comment:
Same point regarding `str()` here - `RequestWithSources.request_id` is `Optional[str]`, so this would log the string `'None'` in case `t.rws.request_id` is `None`. Definitely not an issue in this case since it's just logging, but as a suggestion I think the logs might be clearer if we do something like:

```suggestion
           state.name, transfertool, external_host, external_id, ', '.join((t.rws.request_id or 'MISSING REQUEST ID') for t in transfers))
```

e.g. let's say all request IDs are missing - if all of them are `None` it's not as easy to figure out that the `None` are request IDs, since the first logger message just calls them "transfers". If we call them "missing request IDs", it's a bit clearer.